src/overloads/arrays.jl

```diff
@@ -0,0 +1,208 @@
+
+## Determinant
+LinearAlgebra.det(A::AbstractMatrix{T}) where {T<:AbstractTracer} = second_order_or(A)
+LinearAlgebra.logdet(A::AbstractMatrix{T}) where {T<:AbstractTracer} = second_order_or(A)
+function LinearAlgebra.logabsdet(A::AbstractMatrix{T}) where {T<:AbstractTracer}
+    t = second_order_or(A)
+    return (t, t)
+end
+
+# Fix for issue #108
+function LinearAlgebra.det(A::AbstractMatrix{D}) where {D<:Dual}
+    P, T = split_dual_array(A)
+    p = LinearAlgebra.logdet(P)
+    t = LinearAlgebra.logdet(T)
+    return D(p, t)
+end
+function LinearAlgebra.logdet(A::AbstractMatrix{D}) where {D<:Dual}
+    P, T = split_dual_array(A)
+    p = LinearAlgebra.logdet(P)
+    t = LinearAlgebra.logdet(T)
+    return D(p, t)
+end
+function LinearAlgebra.logabsdet(A::AbstractMatrix{D}) where {D<:Dual}
+    P, T = split_dual_array(A)
+    p1, p2 = LinearAlgebra.logabsdet(P)
+    t1, t2 = LinearAlgebra.logabsdet(T)
+    return (D(p1, t1), D(p2, t2))
+end
+
+## Norm
+function LinearAlgebra.norm(A::AbstractArray{T}, p::Real=2) where {T<:AbstractTracer}
+    return second_order_or(A)
+end
+function LinearAlgebra.opnorm(A::AbstractArray{T}, p::Real=2) where {T<:AbstractTracer}
+    return first_order_or(A)
+end
+function LinearAlgebra.opnorm(A::AbstractMatrix{T}, p::Real=2) where {T<:AbstractTracer}
+    return first_order_or(A)
+end
+
+## Eigenvalues
+
+function LinearAlgebra.eigmax(
+    A::Union{T,AbstractMatrix{T}}; permute::Bool=true, scale::Bool=true
+) where {T<:AbstractTracer}
+    return second_order_or(A)
+end
+function LinearAlgebra.eigmin(
+    A::Union{T,AbstractMatrix{T}}; permute::Bool=true, scale::Bool=true
+) where {T<:AbstractTracer}
+    return second_order_or(A)
+end
+function LinearAlgebra.eigen(
+    A::AbstractMatrix{T};
+    permute::Bool=true,
+    scale::Bool=true,
+    sortby::Union{Function,Nothing}=nothing,
+) where {T<:AbstractTracer}
+    LinearAlgebra.checksquare(A)
+    n = size(A, 1)
+    t = second_order_or(A)
+    values = Fill(t, n)
+    vectors = Fill(t, n, n)
+    return LinearAlgebra.Eigen(values, vectors)
+end
+
+## Inverse
+function LinearAlgebra.inv(A::StridedMatrix{T}) where {T<:AbstractTracer}
+    LinearAlgebra.checksquare(A)
+    t = second_order_or(A)
+    return Fill(t, size(A)...)
+end
+function LinearAlgebra.pinv(
+    A::AbstractMatrix{T}; atol::Real=0.0, rtol::Real=0.0
+) where {T<:AbstractTracer}
+    n, m = size(A)
+    t = second_order_or(A)
+    return Fill(t, m, n)
+end
+
+## Division
+function LinearAlgebra.:\(
+    A::AbstractMatrix{T}, B::AbstractVecOrMat
+) where {T<:AbstractTracer}
+    Ainv = LinearAlgebra.pinv(A)
+    return Ainv * B
+end
+
+## Exponent
+function LinearAlgebra.exp(A::AbstractMatrix{T}) where {T<:AbstractTracer}
+    LinearAlgebra.checksquare(A)
+    n = size(A, 1)
+    t = second_order_or(A)
+    return Fill(t, n, n)
+end
+
+## Matrix power
+function LinearAlgebra.:^(A::AbstractMatrix{T}, p::Integer) where {T<:AbstractTracer}
```

Comment:
here it might be worth specializing on the value of `p`, I assume powers `0` and `1` are common